Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in accessibility code

cdumez opened the pull request:
#### 87333b32fe9c8aec5826c1d4eb2728337556be01
<pre>
Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in accessibility code
<a href="https://bugs.webkit.org/show_bug.cgi?id=285237">https://bugs.webkit.org/show_bug.cgi?id=285237</a>

Reviewed by Darin Adler.

* Source/WebCore/accessibility/AXSearchManager.cpp:
(WebCore::AXSearchManager::findMatchingRange):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::initializeRoleMap):
* Source/WebCore/accessibility/AccessibilityTable.cpp:
(WebCore::AccessibilityTable::isDataTable const):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(convertMathPairsToNSArray):
(createAccessibilitySearchKeyMap):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(WebTransformCGPathToNSBezierPath):

Canonical link: <a href="https://commits.webkit.org/288331@main">https://commits.webkit.org/288331@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fd972cf0d5a8e3d1d5a0388a832ec5ad72a2ff0a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82537 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36638 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87662 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33598 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84635 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2264 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10093 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64324 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22090 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85598 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1629 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75089 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44601 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1525 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29277 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32632 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72816 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29930 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89025 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9835 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7102 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72731 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10063 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70903 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71947 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16089 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/15092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1223 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12807 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9788 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15309 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9662 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13128 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11432 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -2637,139 +2635,139 @@ static void initializeRoleMap()
-        { "deletion"_s, AccessibilityRole::Deletion },
-        { "dialog"_s, AccessibilityRole::ApplicationDialog },
-        { "directory"_s, AccessibilityRole::Directory },
+    const std::array roles {
```
Can't be `constexpr`?

cdumez replied:
Didn't work:
```
/Volumes/Work/WebKit/OpenSource/Source/WebCore/accessibility/AccessibilityObject.cpp:2638:26: error: constexpr variable cannot have non-literal type 'const std::array<RoleEntry, 1 + sizeof...(_Args)>' (aka 'const array<WebCore::RoleEntry, 1 + sizeof...(_Args)>')
 2638 |     constexpr std::array roles {
```

darinadler commented on Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
```diff
@@ -827,53 +825,53 @@ - (NSString *)outerHTML
-        { NSAccessibilityUnderlineSearchKey, AccessibilitySearchKey::Underline },
-        { NSAccessibilityUnvisitedLinkSearchKey, AccessibilitySearchKey::UnvisitedLink },
-        { NSAccessibilityVisitedLinkSearchKey, AccessibilitySearchKey::VisitedLink }
+    const std::array searchKeys {
```
Can't be `constexpr`?

cdumez replied:
Didn't work for this one.

cdumez committed:
Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in accessibility code
https://bugs.webkit.org/show_bug.cgi?id=285237

Reviewed by Darin Adler.

* Source/WebCore/accessibility/AXSearchManager.cpp:
(WebCore::AXSearchManager::findMatchingRange):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::initializeRoleMap):
* Source/WebCore/accessibility/AccessibilityTable.cpp:
(WebCore::AccessibilityTable::isDataTable const):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(convertMathPairsToNSArray):
(createAccessibilitySearchKeyMap):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(WebTransformCGPathToNSBezierPath):

Canonical link: https://commits.webkit.org/288331@main
